### ISSUE TITLE
Numpy: errors inputting numeric type aliases to `np.iinfo` and `np.finfo`

### ISSUE THREAD

**Issue by bersbersbers** [microsoft/pyright#6559]
**Describe the bug**
I do not have public source code for you, but I learned how to install `pyright` from source (thanks to your instructions on https://microsoft.github.io/pyright/#/build-debug?id=building-the-cli), so I am testing the GitHub version early for #6523 and to find further regressions. Turns out, I found one - here's three examples:

**Code or Screenshots**
```python
from typing import Any

import numpy as np

np.iinfo(np.int32)

Int = np.integer[Any]
Float = np.floating[Any]
Number = Int | Float
TYPE_UINT16: type[Number] = np.uint16
if issubclass(TYPE_UINT16, np.integer):  # pyright: ignore[reportUnnecessaryIsInstance]
    np.iinfo(TYPE_UINT16)

TYPE_FLOAT64 = np.float64
np.finfo(TYPE_FLOAT64)
```

**VS Code extension or command-line**

`node index.js` after building from GitHub, revision 60e4050.

**Comment by erictraut** [comment]
Thanks for the bug report. This was a long-standing bug that was exposed by a recent change. It also exposed a hole in our unit tests, which I've addressed along with the bug fix. And surprisingly, none of the existing mypy_primer source bases were affected by this issue.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.338, which I just published. It will also be included in a future release of pylance.
